apps/server/src/modules/user/service/user.service.ts

```diff
@@ -280,4 +280,15 @@ export class UserService implements DeletionService, IEventHandler<UserDeletedEv
 
 		return DomainDeletionReportBuilder.build(DomainName.CALENDAR, extractedOperationReport);
 	}
+
+	public async isEmailFromExternalSourceUnique(email: string, externalId?: string): Promise<boolean> {
```

Comment:
please rename to isEmailUniqueForExternal.

Do you think its valid to pass a undefined externalId here? If yes think about early return for more readability